cmd/dep/root_analyzer.go

```diff
@@ -124,17 +127,32 @@ func (a *rootAnalyzer) DeriveManifestAndLock(dir string, pr gps.ProjectRoot) (gp
 }
 
 func (a *rootAnalyzer) FinalizeRootManifestAndLock(m *dep.Manifest, l *dep.Lock) {
-	// Remove dependencies from the manifest that aren't used
-	for pr := range m.Constraints {
-		var used bool
-		for _, y := range l.Projects() {
-			if pr == y.Ident().ProjectRoot {
-				used = true
-				break
+	if a.gopath {
```

Comment:
I think both paths of this if/else are safe to run unconditionally. If a direct dep was added during solve, either because the imported config was incomplete or due to "network solving", then it's appropriate to add it to the manifest. Same for removing unused dependencies from the manifest.